superset/utils/log.py

```diff
@@ -36,58 +36,69 @@ def log(
     ) -> None:
         pass
 
-    def log_this(self, f: Callable[..., Any]) -> Callable[..., Any]:
+    def log_with_context(
```

Comment:
Eventually we can add a context manager (allowing the `with` clause in python) so that we can do things like:

```python
with log_this_with_duration(action="do_stuff"):
    do_stuff_that_takes_long()
```

Reply:
example here for something similar for statsd logging https://github.com/apache/incubator-superset/blob/master/superset/utils/decorators.py#L38-L47

Reply:
Didn't know you can do something like this. Thanks for the pointer!

Reply:
Refactored with `contextmanager`. Didn't use `contextlib2` since it seems to be a backport package for older pythons (<3.6) which we no longer support. Plus we also didn't use the advanced stuff it was trying to backport.